src/VecSim/vec_sim_tiered_index.h

```diff
@@ -96,8 +96,15 @@ class VecSimTieredIndex : public VecSimIndexInterface {
     static VecSimWriteMode getWriteMode() { return VecSimIndexInterface::asyncWriteMode; }
 
 #ifdef BUILD_TESTS
-    inline VecSimIndexAbstract<DistType> *getFlatBufferIndex() { return this->frontendIndex; }
+    inline VecSimIndexAbstract<DataType, DistType> *getFlatBufferIndex() {
+        return this->frontendIndex;
+    }
     inline size_t getFlatBufferLimit() { return this->flatBufferLimit; }
+
+    virtual void fitMemory() {
```

Comment:
isn't it being used in testing? can we get it covered?

Reply:
only in the benchmark
i can write a unit test